### PR TITLE
Archive group access limited

### LINKED ISSUE #525: Archive group access limited
Currently:

- Anybody can add people to a group (should be only author and admins)
- Nobody can archive group (should be the same)

Screenshot 2024-02-19 at 20 06 35

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

-   [x] Bug Fix

## Related issues

Resolve #525

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/crew/assets/81121562/40d7601f-d529-4af4-bac8-0555956f86a5

File: src/components/ArchiveGroup/ArchiveGroup.i18n/ru.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "Cancel": "",
+    "Archive": "",
```
Comment (awinogradov) [inline src/components/ArchiveGroup/ArchiveGroup.i18n/ru.json:3]: ?

File: src/components/ArchiveGroup/ArchiveGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    Form,
+    FormAction,
+    FormActions,
+    FormTitle,
+    Modal,
+    ModalContent,
+    ModalCross,
+    ModalHeader,
+} from '@taskany/bricks';
+import { Group } from 'prisma/prisma-client';
```
Comment (awinogradov) [inline src/components/ArchiveGroup/ArchiveGroup.tsx:12]: It's restricted already.

File: src/components/TeamProfilePreview/TeamProfilePreview.tsx [hunk 3]
```diff
@@ -60,7 +69,35 @@ const TeamProfilePreview = ({ groupId }: UserProps): JSX.Element => {
...
+
+                            <ArchiveGroupModal
+                                visible={archiveGroupModalVisibility.value}
+                                group={group}
+                                onClose={archiveGroupModalVisibility.setFalse}
+                            />
+                            {user?.role === UserRole.ADMIN &&
+                                (childrenQuery.data?.length ? (
+                                    <>
+                                        <div ref={ref}>
```
Comment (awinogradov) [inline src/components/TeamProfilePreview/TeamProfilePreview.tsx:81]: Why not put ref on InlineTrigger?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/components/TeamProfilePreview/TeamProfilePreview.tsx (hunk 3 above)
Comment (Katrin-kudryash) [reply src/components/TeamProfilePreview/TeamProfilePreview.tsx:81]: If InlineTrigger has a disabled prop, then popup is not visible

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/crew/assets/81121562/fcb068ed-61fa-46a8-9e12-66545066eae3

File: src/components/TeamProfilePreview/TeamProfilePreview.tsx [hunk 4]
```diff
@@ -60,7 +78,35 @@ const TeamProfilePreview = ({ groupId }: UserProps): JSX.Element => {
...
                         <NarrowSection>
                             <TransferGroupForm group={group} />
-                            <InlineTrigger icon={<IconBinOutline size="xs" />} text={tr('Archive group')} disabled />
+
+                            <ArchiveGroupModal
+                                visible={archiveGroupModalVisibility.value}
+                                group={group}
+                                onClose={archiveGroupModalVisibility.setFalse}
+                            />
+                            {user?.role === UserRole.ADMIN &&
```
Comment (awinogradov) [inline src/components/TeamProfilePreview/TeamProfilePreview.tsx:87]: nullable

**Review by tsumo — APPROVED** [review]

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/crew/assets/81121562/3863e67e-0509-4e1e-9421-10c02ba75073

File: src/components/GroupListItem.tsx [hunk 5]
```diff
@@ -20,14 +20,14 @@ const StyledWrapper = styled.div`
     flex-wrap: nowrap;
 `;
 
-export const GroupListItem = ({ group }: GroupListItemProps) => {
+export const GroupListItem = ({ groupId, groupName, ...props }: GroupListItemProps) => {
```
Comment (awinogradov) [inline src/components/GroupListItem.tsx:23]: Unnecessary spread

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Pull request merged**